core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -48,25 +57,34 @@ import whisk.core.entity.WhiskActivation
 
 class LoadBalancerService(config: WhiskConfig, verbosity: LogLevel)(
     implicit val actorSystem: ActorSystem)
-    extends LoadBalancerToKafka
-    with Logging {
+    extends Logging {
 
+    /** The execution context for futures */
     implicit val executionContext: ExecutionContext = actorSystem.dispatcher
 
     /**
      * The two public methods are getInvokerHealth and the inherited doPublish methods.
      */
     def getInvokerHealth(): JsObject = invokerHealth.getInvokerHealthJson()
 
-    override def getInvoker(message: ActivationMessage): Option[Int] = invokerHealth.getInvoker(message)
+    /**
+     * Gets an invoker index to send request to.
+     *
+     * @return index of invoker to receive request
+     */
+    def getInvoker(message: ActivationMessage): Option[Int] = invokerHealth.getInvoker(message)
 
-    override val producer = new KafkaProducerConnector(config.kafkaHost, executionContext)
+    /** Gets a producer which can publish messages to the kafka bus. */
+    val producer = new KafkaProducerConnector(config.kafkaHost, executionContext)
 
     private val invokerHealth = new InvokerHealth(config, resetIssueCountByInvoker, () => producer.sentCount())
 
-    // map stores the promise which either completes if an active response is received or
-    // after the set timeout expires
-    private val queryMap = new TrieMap[ActivationId, Promise[WhiskActivation]]
+    /**
+     * A map storing current activations based on ActivationId.
+     * The promise value represents the obligation of writing the answer back.
+     */
+    type ActivationEntry = (Instant, Promise[WhiskActivation])
```

Comment:
Would it make sense to make this a proper `case class` instead of an aliased tuple?